app/gui2/src/components/GraphEditor/widgets/WidgetTableEditor/tableNewArgument.ts

```diff
@@ -0,0 +1,228 @@
+ * only values actually used by the composable) */
+interface ColumnDef {
+  colId?: string
+  headerName: string
+  valueGetter: ({ data }: { data: RowData | undefined }) => any
+  valueSetter: ({ data, newValue }: { data: RowData; newValue: any }) => boolean
+  headerComponentParams?: HeaderParams
+}
+
+namespace cellValueConversion {
+  export function astToAgGrid(ast: Ast.Ast) {
+    if (ast instanceof Ast.TextLiteral) return ast.rawTextContent
+    else if (ast instanceof Ast.NumericLiteral) return parseFloat(ast.code())
+    else return undefined
+  }
+
+  export function agGridToAst(value: unknown, module: MutableModule): Ast.Owned {
+    if (value == null) {
+      return Ast.TextLiteral.new('', module)
+    }
+    return (
+      Ast.NumericLiteral.tryParse(`${value}`, module) ?? Ast.TextLiteral.new(`${value}`, module)
+    )
+  }
+}
+
+/**
+ * A composable responsible for interpreting `Table.new` expressions, creating AGGrid column
+ * definitions allowing also editing AST through AGGrid editing.
+ *
+ * @param input the widget's input
+ * @param graph the graph store
+ * @param onUpdate callback called when AGGrid was edited by user, resulting in AST change.
+ */
+export function useTableNewArgument(
+  input: ToValue<WidgetInput & { value: Ast.Ast }>,
+  graph: { startEdit(): Ast.MutableModule },
+  onUpdate: (update: WidgetUpdate) => void,
+) {
+  const columnsAst = computed(() => {
+    const inputAst = toValue(input).value
+    if (!(inputAst instanceof Ast.App)) return undefined
+    if (!(inputAst.argument instanceof Ast.Vector)) return undefined
+    return inputAst.argument
+  })
+
+  const columns = computed(() => {
+    const cols = columnsAst.value?.values()
+    if (cols == null) return []
+    const arr = Array.from(filterDefined(mapIterator(cols, (element) => readColumn(element))))
+    if (arr.some((x) => x == null)) return []
+    return arr
+  })
+
+  const rowCount = computed(() =>
+    columns.value.reduce((soFar, col) => Math.max(soFar, col.data.length), 0),
+  )
+
+  const undersizedColumns = computed(() =>
+    columns.value.filter((col) => col.data.length < rowCount.value),
+  )
+
+  function readColumn(ast: Ast.Ast): { name: Ast.TextLiteral; data: Ast.Vector } | undefined {
+    if (!(ast instanceof Ast.Vector)) return undefined
+    const elements = ast.values()
+    const first = elements.next()
+    if (first.done) return undefined
+    const second = elements.next()
+    if (second.done) return undefined
+    if (!elements.next().done) return undefined
+
+    if (!(first.value instanceof Ast.TextLiteral)) return undefined
+    if (!(second.value instanceof Ast.Vector)) return undefined
+    return { name: first.value, data: second.value }
+  }
+
+  function fixColumns(edit: Ast.MutableModule) {
+    for (const column of undersizedColumns.value) {
+      const data = edit.getVersion(column.data)
+      while (data.length < rowCount.value) {
+        data.push(Ast.TextLiteral.new('', edit))
```

Comment:
The parser used by `''.to Table` produces `Nothing` values when encountering empty cells